dist/contour.js

```diff
@@ -336,12 +336,37 @@
                 var dataMax = max != null ? max : _.max(domain);
                 ticks = ticks == null ? 5 : ticks;
                 var niceMinMax = axisHelpers.niceMinMax(dataMin, dataMax, ticks, zeroAnchor);
-                return [ niceMinMax.min, niceMinMax.max ];
+                // return [niceMinMax.min, niceMinMax.max];
+                // // we want null || undefined for all this comparasons
+                // // that == null gives us
+                if (min == null && max == null) {
+                    return [ niceMinMax.min, niceMinMax.max ];
+                }
+                if (min == null) {
+                    return [ Math.min(niceMinMax.min, max), max ];
+                }
+                if (max == null) {
+                    return [ min, Math.max(min, niceMinMax.max) ];
+                }
+                return [ min, max ];
             },
-            niceTicks: function(min, max, ticks, zeroAnchor) {
+            niceTicks: function(min, max, ticks, zeroAnchor, opts) {
                 ticks = ticks == null ? 5 : ticks;
                 var niceMinMax = axisHelpers.niceMinMax(min, max, ticks, zeroAnchor);
-                return niceMinMax.tickValues;
+                var tickValues = niceMinMax.tickValues;
+                // ensure that y-axis endpoints are labelled
+                if (opts.min != null && opts.max != null) {
+                    tickValues.push(min);
+                    tickValues.push(max);
+                } else if (opts.min != null) {
+                    tickValues.push(min);
+                } else if (opts.max != null) {
+                    tickValues.push(max);
```

Comment:
I'm not sure if this will be an issue, actually - the tickValues array will already be populated (by the niceMinMax function), and so this check just prevents a null value from being added to that array.

Agreed on both comments below though - I'll fix up those lines.